hooks/tk-multi-publish2/basic/publish_distortion_maps.py

```diff
@@ -0,0 +1,340 @@
+        if not publish_template:
+            self.logger.debug(
+                "The valid publish template could not be determined for the "
+                "session item. Not accepting the item."
+            )
+            accepted = False
+
+        if accepted:
+            # because a publish template is configured, disable context change. This
+            # is a temporary measure until the publisher handles context switching
+            # natively.
+            item.context_change_allowed = False
+
+        return { "accepted": accepted, "checked": accepted }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        publisher = self.parent
+        engine: SynthEyesEngine = publisher.engine
+        hlev = engine.get_syntheyes_connection()
+               
+        # ---- ensure the session has been saved
+        
+        # get the normalized path
+        path = sgtk.util.ShotgunPath.normalize(hlev.SNIFileName())
+        if not path or hlev.HasChanged():
+            # the session still requires saving. provide a save button.
+            # validation fails.
+            error_msg = "The SynthEyes session has not been saved."
+            self.logger.error(error_msg, extra=_get_save_as_action())
+            raise Exception(error_msg)
+
+        item.properties["publish_type"] = "SynthEyes Distortion Maps"
+        template_id = settings["Publish Template"].value
+        publish_template = publisher.get_template_by_name(template_id)
+        item.properties["publish_template"] = publish_template
+        
+        # get the configured work file template
+        work_template = item.parent.properties.get("work_template")
+
+        # get the current scene path and extract fields from it using the work template:
+        work_fields = work_template.get_fields(path)
+        # set the export_name for syntheyes required to resolve the publish path
+        item_name = item.name.replace(" ", "_")
+        work_fields["syntheyes.export_name"] = item_name
+
+        # ensure the fields work for the publish template
+        missing_keys = publish_template.missing_keys(work_fields)
+        if missing_keys:
+            error_msg = (
+                "Work file '%s' missing keys required for the "
+                "publish template: %s" % (path, missing_keys)
+            )
+            raise Exception(error_msg)
+
+        # resolve undistort and redistort template strings
+        for template_id, prop_name in {"Undistort Template" : "undistort_name", "Redistort Template": "redistort_name"}.items():
```

Comment:
Das ist ein bisschen unschön :D

Reply:
Ich meinte eigentlich die Initialisierung der Dicts in for und nicht als konstate Variable getrennt. Einfach für Wartbarkeit, ist aber nichts Schlimmes :)